src/webview.rs

```diff
@@ -286,4 +292,18 @@ impl Window {
         }
         false
     }
+
+    fn allow_navigation(&self, url: ServoUrl) -> crate::Result<bool> {
+        if let Some(ref sender) = *self.event_listeners.on_navigation_starting.lock().unwrap() {
+            let (result_sender, receiver) =
+                ipc::channel::<bool>().map_err(|e| ipc::IpcError::Io(e))?;
+            sender
+                .send((url.into_url(), result_sender))
+                .map_err(|e| ipc::IpcError::Bincode(e))?;
+            let result = receiver.recv()?;
+            Ok(result)
+        } else {
+            Ok(true)
+        }
```

Comment:
I still believe there needs to be a dedicated channel for communicating asynchronously. The `lock` and `recv` here are strong deadlock material.

Reply:
I can clone the sender out here to prevent that in terms of dead lock, we can do this asynchronously though if you want it that way

About a dedicated channel, do you mean an unified sender for all info we need to send to the core? I don't think it works well here to be honest